pkg/platform/kube/platform_test.go

```diff
@@ -118,6 +124,257 @@ func (suite *KubePlatformTestSuite) TestFunctionTriggersEnriched() {
 	}
 }
 
+type APIGatewayKubePlatformTestSuite struct {
+	KubePlatformTestSuite
+}
+
+func (suite *APIGatewayKubePlatformTestSuite) TestAPIGatewayEnrichmentAndValidation() {
+
+	for _, testCase := range []struct {
+		apiGatewayConfig           platform.APIGatewayConfig
+		expectedEnrichedAPIGateway platform.APIGatewayConfig
+
+		// keep empty when shouldn't fail
+		validationError string
+	}{
+
+		// test spec.name enriched from meta.name
+		{
+			apiGatewayConfig: func() platform.APIGatewayConfig {
+				apiGatewayConfig := suite.compileAPIGatewayConfig()
+				apiGatewayConfig.Spec.Name = ""
+				apiGatewayConfig.Meta.Name = "meta-name"
+				return apiGatewayConfig
+			}(),
+			expectedEnrichedAPIGateway: func() platform.APIGatewayConfig {
+				apiGatewayConfig := suite.compileAPIGatewayConfig()
+				apiGatewayConfig.Spec.Name = "meta-name"
+				apiGatewayConfig.Meta.Name = "meta-name"
+				return apiGatewayConfig
+			}(),
+			validationError: "",
+		},
+
+		// test meta.name enriched from spec.name
+		{
+			apiGatewayConfig: func() platform.APIGatewayConfig {
+				apiGatewayConfig := suite.compileAPIGatewayConfig()
+				apiGatewayConfig.Spec.Name = "spec-name"
+				apiGatewayConfig.Meta.Name = ""
+				return apiGatewayConfig
+			}(),
+			expectedEnrichedAPIGateway: func() platform.APIGatewayConfig {
+				apiGatewayConfig := suite.compileAPIGatewayConfig()
+				apiGatewayConfig.Spec.Name = "spec-name"
+				apiGatewayConfig.Meta.Name = "spec-name"
+				return apiGatewayConfig
+			}(),
+			validationError: "",
+		},
+
+		// test enrichment of waiting for provisioning state
+		{
+			apiGatewayConfig: func() platform.APIGatewayConfig {
+				apiGatewayConfig := suite.compileAPIGatewayConfig()
+				return apiGatewayConfig
+			}(),
+			expectedEnrichedAPIGateway: func() platform.APIGatewayConfig {
+				apiGatewayConfig := suite.compileAPIGatewayConfig()
+				apiGatewayConfig.Status.State = platform.APIGatewayStateWaitingForProvisioning
+				return apiGatewayConfig
+			}(),
+			validationError: "",
+		},
+
+		// test namespace existence
+		{
+			apiGatewayConfig: func() platform.APIGatewayConfig {
+				apiGatewayConfig := suite.compileAPIGatewayConfig()
+				apiGatewayConfig.Meta.Namespace = ""
+				return apiGatewayConfig
+			}(),
+			expectedEnrichedAPIGateway: func() platform.APIGatewayConfig {
```

Comment:
do you need the expected here? because you can simply fail on the validation and not check the enrichment as it has no meaning when an error occured